components/dashboard/UpcomingInterviewDialog/formValidationHook.js

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from 'react';
+
+const useFormValidation = (initialState, validate, runOnSubmit) => {
+  const [values, setValues] = useState(initialState);
+  const [errors, setErrors] = useState({});
+
+  const [touched, setTouched] = useState([]);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      const noErrors = Object.keys(errors).length === 0;
+      if (noErrors) {
+        setTouched([]);
+        runOnSubmit(values);
+        setSubmitting(false);
+      } else {
+        setSubmitting(false);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [errors]);
+
+  useEffect(() => {
+    const validationErrors = validate(values);
+    const touchedErrors = Object.keys(validationErrors)
+      .filter(key => touched.includes(key))
+      .reduce((acc, key) => {
+        if (!acc[key]) {
+          acc[key] = validationErrors[key];
+        }
+        return acc;
+      }, {});
+    setErrors(touchedErrors);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can we find a solution here that doesn't require breaking the rules? I'm leery of it being a slippery slope and leading to hard-to-trace bugs.

I haven't looked closely at this file, but seems like `useEffect` may not be needed for this (validation could be performed in the render cycle).

Reply:
I can work on refactoring this